lib/std/fs/file.zig

```diff
@@ -385,6 +385,274 @@ pub const File = struct {
+        .windows => struct {
+            attributes: windows.DWORD,
+            reparse_tag: windows.DWORD,
+            _size: u64,
+            atime: i128,
+            mtime: i128,
+
+            const Self = @This();
+
+            /// Returns the size of the file
+            pub fn size(self: Self) u64 {
+                return self._size;
+            }
+
+            /// Returns a `Permissions` struct, representing the permissions on the file
+            pub fn permissions(self: Self) Permissions {
+                return Permissions{ .attributes = self.attributes };
+            }
+
+            /// Returns the `Kind` of the file.
+            /// On Windows, returns: `.File`, `.Directory`, `.SymLink` or `.Unknown`
+            pub fn kind(self: Self) Kind {
+                if (self.attributes & windows.FILE_ATTRIBUTE_REPARSE_POINT != 0) {
+                    if (self.reparse_tag & 0x20000000 != 0) {
+                        return .SymLink;
+                    }
+                } else if (self.attributes & windows.FILE_ATTRIBUTE_DIRECTORY != 0) {
+                    return .Directory;
+                } else {
+                    return .File;
+                }
+                return .Unknown;
+            }
+
+            /// Returns the last time the file was accessed in nanoseconds since UTC 1970-01-01
+            pub fn accessed(self: Self) i128 {
+                return self.atime;
+            }
+
+            /// Returns the time the file was modified in nanoseconds since UTC 1970-01-01
+            pub fn modified(self: Self) i128 {
+                return self.mtime;
+            }
+        },
+        else => struct {
+            stat: Stat,
+
+            const Self = @This();
+
+            /// Returns the size of the file
+            pub fn size(self: Self) u64 {
+                return self.stat.size;
+            }
+
+            /// Returns a `Permissions` struct, representing the permissions on the file
+            pub fn permissions(self: Self) Permissions {
+                return Permissions{ .mode = self.stat.mode };
+            }
+
+            /// Returns the `Kind` of file.
+            /// On Windows, returns: `.File`, `.Directory`, `.SymLink` or `.Unknown`
+            pub fn kind(self: Self) Kind {
+                return self.stat.kind;
+            }
+
+            /// Returns the last time the file was accessed in nanoseconds since UTC 1970-01-01
+            pub fn accessed(self: Self) i128 {
+                return self.stat.atime;
+            }
+
+            /// Returns the time the file was modified in nanoseconds since UTC 1970-01-01
+            pub fn modified(self: Self) i128 {
+                return self.stat.mtime;
+            }
+        },
+    };
+
+    pub const MetadataError = StatError;
+
+    /// Returns a `Metadata` struct, representing the permissions on the file
+    pub fn metadata(self: File) MetadataError!Metadata {
```

Comment:
How about `getMetadata`?  I think Zig's std convention is usually to use verbs for getting/setting things?

Reply:
When discussing this in the Discord server, @ifreund said to me the opposite -- that I should drop the 'get' prefix which I was using previously. That matches with `.allocator()` too.